docs/what-just-happened/what-just-happened_test_plan.md

```diff
@@ -0,0 +1,145 @@
+# what-just-happened Test Plan
+
+## Table of Contents
+- [Introduction](#introduction)
+- [Pre test](#pre-test)
+- [Test Plan](#test-plan)
+    - [Overview](#overview)
+    - [Test Cases](#test-cases)
+        - [L2 Drops](#l2-drops)
+        - [L3 Drops](#l3-drops)
+
+
+## Introduction
+what-just-happened is a feature intended to track lost packets and display the drop reason.
+
+The purpose of this test is:
+
+1. Check that wrong packets are dropped. 
+2. Check that the description in 'show what-just-happened' command output suits the dropped packet.
+ 
+The test will focus on L2, L3 raw packets drops. 
+
+The implementation will be done using PyTest framework.
+
+## Pre test
+Before all test cases execution, the test should check whether what-just-happened feature is available and enabled. 
+
+For this purpose, a pre test fixture will do the following:
+
+- Execute ```show feature``` command on DUT. 
+- Parse output table.
+- Check if 'what-just-happened' appears in the table (availability), and if it is enabled.
+
+```bash
+Feature             Status
+------------------  --------
+telemetry           enabled
+sflow               disabled
+what-just-happened  enabled
+```
+
+- If one of the checks on previous phase is indicated what-just-happened as not available, skip the test. 
+
+## Test plan
+
+### Overview
+The test will use test cases appear in a file named drop_packets.py.
+
+This file contains creation of packets intend to drop.
+
+for example: 
+
+```test_equal_smac_dmac_drop``` function, generates a TCP packet which equal source mac and destination mac addresses.
+
+After each test case generates the packet, it calls do_test() function which implemented inside test_what_just_happened.py file. 
+
+do_test() function will send the packet, check if packet indeed dropped, and parse the output of 'show what-just-happened' command. 
```

Comment:
Does the test verifies that there is only one signle entry in the table? E.g. does running background traffic affects the test result?